synapse/handlers/federation.py

```diff
@@ -830,6 +830,37 @@ def maybe_backfill(self, room_id, current_depth):
             logger.debug("Not backfilling as no extremeties found.")
             return
 
+        # We only want to paginate if we can actually see the events we'll get,
+        # as otherwise we'll just spend a lot of resources to get redacted
+        # events.
+        #
+        # We do this by filtering all the extremities and seeing if any remain.
```

Comment:
surely it's possible for me to be able to see some earlier events in a room even if I can't see the extremities' successors? For instance, if history vis was changed at some point?